From afa537df091fe52a944df5865c3e34ff815569bc Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <ross.lagerwall@citrix.com>
Date: Fri, 6 May 2016 15:04:50 +0100
Subject: [PATCH 2/5] sched: Remove dependency on __LINE__ for release builds

When using live patching, use of __LINE__ can generate spurious changes in
functions due to embedded line numbers.  For release builds, remove the
use of these line numbers in domain_crash*() and print the current text
address instead.

Signed-off-by: Ross Lagerwall <ross.lagerwall@citrix.com>
diff --git a/xen/include/xen/sched.h b/xen/include/xen/sched.h
index 037c83fda2..3948c937e1 100644
--- a/xen/include/xen/sched.h
+++ b/xen/include/xen/sched.h
@@ -814,11 +814,18 @@ void vcpu_end_shutdown_deferral(struct vcpu *v);
  * from any processor.
  */
 void __domain_crash(struct domain *d);
+#if defined(NDEBUG) && defined(CONFIG_LIVEPATCH)
+#define print_domain_crash() \
+    printk(XENLOG_ERR "domain_crash called from %pS\n", current_text_addr());
+#else
+#define print_domain_crash() \
+    printk(XENLOG_ERR "domain_crash called from %s:%d\n", __FILE__, __LINE__);
+#endif
+
 #define domain_crash(d, ...)                                        \
     do {                                                            \
         if ( count_args(__VA_ARGS__) == 0 )                         \
-            printk(XENLOG_ERR "domain_crash called from %s:%d\n",   \
-                   __FILE__, __LINE__);                             \
+            print_domain_crash();                             \
         else                                                        \
             printk(XENLOG_ERR __VA_ARGS__);                         \
         __domain_crash(d);                                          \
